packages/remix-dev/compiler/watch.ts

```diff
@@ -29,7 +29,7 @@ export type WatchOptions = Partial<CompileOptions> & {
 };
 
 export async function watch(
-  config: RemixConfig,
+  configGetter: RemixConfig | (() => RemixConfig | Promise<RemixConfig>),
```

Comment:
Could you separate out the initial config from getter for the subsequent configs? That will make the types simpler and won't require any `typeof` checks further on.

```suggestion
  initialConfig: RemixConfig,
  reloadConfig: (rootDirectory: string) => RemixConfig = readConfig,
```

Also note that the root directory _could_ change when the config is reloaded, so to be extra defensive, its nice to accept the current `rootDirectory` as an arg.

Reply:
nit: I also think `reloadConfig` conveys that this is doing some I/O so I prefer that to `getNewConfig`

Reply:
>Could you separate out the initial config from getter for the subsequent configs? That will make the types simpler and won't require any typeof checks further on.

I'm ok with the change but this would mean `watch` now has 2 optional parameters: `watch(initialConfig, reloadConfig?, options?)` so it could potentially go `watch(config, undefined, {...})`. It's also a breaking change for anyone using `watch` already. Should we move `reloadConfig` to the existing options object instead? Just want to double check with you and @jacob-ebey before updating the PR 👍

Reply:
yea including it in `options` sounds good

Reply:
Alright, I've updated the PR, thanks!